docs_src/microservices/device/V3Migration.md

```diff
@@ -8,29 +8,145 @@ See [Top Level V3 Migration Guide](../../../V3TopLevelMigration) for details app
 
 ### Device Files
 
-TBD
+1. Change device definition file to YAML format.
+2. Remove `LastConnected` and `LastReported` configs.
+3. ProtocolProperties now supports typed values.
+
+    !!! example "ProtocolProperty with typed values"
+    
+        ```yaml
+        protocols:
+          other:
+            Address: simple01
+            Port: 300
+        ```
 
 ### Device Profile Files
 
-TBD
+N/A
 
 ### Provision Watcher files
 
-TBD
+1. ProvisionWatcher can now be added during device service startup by loading the definition files from the `ProvisionWatchersDir` configuration.
+
+    !!! example "Example Configuration"
+    
+        ```yaml
+        Device:
+            ProvisionWatchersDir: ./res/provisionwatchers
+        ```
+
+2. ProvisionWatcher definition file is in YAML format.
+
+    !!! example "Pre-defined ProvisionWatcher"
+
+        ```yaml
+        name: Simple-Provision-Watcher
+        serviceName: device-simple
+        labels:
+          - simple
+        identifiers:
+          Address: simple[0-9]+
+          Port: 3[0-9]{2}
+        blockingIdentifiers:
+          Port:
+            - 397
+            - 398
+            - 399
+        adminState: UNLOCKED
+        discoveredDevice:
+          profileName: Simple-Device
+          adminState: UNLOCKED
+          autoEvents:
+            - interval: 15s
+              sourceName: SwitchButton
+          properties:
+            testPropertyA: weather
+            testPropertyB: meter
+        ```
+
+        Note that in EdgeX 3.0, `ServiceName` is moved out of `DiscoveredDevice` to top level.
```

Comment:
`DiscoveredDevice` is new in EdgeX 3.0, so `ServiceNames` is as is.
Please describe all the Provision Watcher changes:
https://docs.edgexfoundry.org/3.0/V3TopLevelMigration/#core-metadata

Reply:
added all Provision Watcher changes